app/services/feature-flags.js

```diff
@@ -78,6 +83,28 @@ export default Service.extend({
     }
   }).drop(),
 
+  _persistToLocalStorage(feature, status) {
+    const featureState = JSON.parse(this.storage.getItem('travis.features'));
+    const currentUserFeatureState = featureState[this.auth.userName];
+    const idx = currentUserFeatureState.findIndex(f => Object.keys(f)[0] === feature);
+    if (idx !== -1) {
+      currentUserFeatureState.splice(idx, 1);
+    }
+    currentUserFeatureState.pushObject({ [feature]: status });
+    this.storage.setItem('travis.features', JSON.stringify({
+      ...featureState,
+      [this.auth.userName]: currentUserFeatureState
+    }));
+  },
+
+  applyFeatureState(feature) {
+    const features = this.features;
+    let { name, enabled } = feature;
+
+    enabled ? features.enable(name) : features.disable(name);
+    this._persistToLocalStorage(name, enabled);
+  },
+
```

Comment:
I like the idea of moving these functions into this service. I think they make more sense here :+1:

Reply:
Yeah, +1 from me as well 👍